pkg/api/kubegraph/edges.go

```diff
@@ -91,31 +92,36 @@ func AddAllExposedPodEdges(g osgraph.MutableUniqueGraph) {
 	}
 }
 
-// AddManagedByRCPodEdges ensures that a directed edge exists between an RC and all the pods
+// AddManagedByControllerPodEdges ensures that a directed edge exists between a controller and all the pods
 // in the graph that match the label selector
-func AddManagedByRCPodEdges(g osgraph.MutableUniqueGraph, rcNode *kubegraph.ReplicationControllerNode) {
-	if rcNode.Spec.Selector == nil {
+func AddManagedByControllerPodEdges(g osgraph.MutableUniqueGraph, to graph.Node, namespace string, selector map[string]string) {
+	if selector == nil {
 		return
 	}
-	query := labels.SelectorFromSet(rcNode.Spec.Selector)
+	query := labels.SelectorFromSet(selector)
 	for _, n := range g.(graph.Graph).Nodes() {
 		switch target := n.(type) {
 		case *kubegraph.PodNode:
-			if target.Namespace != rcNode.Namespace {
+			if target.Namespace != namespace {
 				continue
 			}
 			if query.Matches(labels.Set(target.Labels)) {
-				g.AddEdge(target, rcNode, ManagedByRCEdgeKind)
+				g.AddEdge(target, to, ManagedByControllerEdgeKind)
 			}
 		}
 	}
 }
 
-// AddAllManagedByRCPodEdges calls AddManagedByRCPodEdges for every ServiceNode in the graph
-func AddAllManagedByRCPodEdges(g osgraph.MutableUniqueGraph) {
+// AddAllManagedByControllerPodEdges calls AddManagedByControllerPodEdges for every node in the graph
+// TODO: should do this through an interface (selects pods)
```

Comment:
This won't do long term.  Things outside the legacy API group manage pods.  In fact, if we kept the API clean, this would be broken now since petsets aren't in the legacy API group.